include/clad/Differentiator/VisitorBase.h

```diff
@@ -191,8 +191,13 @@ namespace clad {
       V.beginBlock();
       func();
       clang::CompoundStmt* body = V.endBlock();
-      clang::Expr* lambda =
-          S.ActOnLambdaExpr(noLoc, body, V.getCurrentScope()).get();
+      #if CLANG_VERSION_MAJOR > 17
+        clang::Expr* lambda =
+            S.ActOnLambdaExpr(noLoc, body).get();
+      #else
+        clang::Expr* lambda =
+            S.ActOnLambdaExpr(noLoc, body, V.getCurrentScope()).get();
+      #endif // CLANG_VERSION_MAJOR > 17
```

Comment:
We could add a clad_compat::ActOnLambdaExpr(S, loc, body, scope) function, but this may lead to unnecessary additional calculation of the last parameter in cases of clang>17. I recommend another approach:
```suggestion
        clang::Expr* lambda = S.ActOnLambdaExpr(noLoc, body/*,*/
          CLAD_COMPAT_CLANG17_ActOnLambdaExpr_getCurrentScope_ExtraParam(V)).get();
```
In Compatibility.h must add this:
```
      #if CLANG_VERSION_MAJOR > 17
        #define CLAD_COMPAT_CLANG17_ActOnLambdaExpr_getCurrentScope_ExtraParam(V) /**/
      #else
        #define CLAD_COMPAT_CLANG17_ActOnLambdaExpr_getCurrentScope_ExtraParam(V) ,(V).getCurrentScope()
      #endif // CLANG_VERSION_MAJOR > 17
```